### PR TITLE
simple macros

### PR DESCRIPTION
**Pull request by bwo**
very simple! `define-macro` looking inside of what's being defined and setting a flag on the `Procedure` is ... I'm not thrilled about it, but it works.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ pub enum Value {
 
     Symbol(intern::Symbol),
     ForeignFn(ForeignFunction<()>),
-    Lambda(Procedure),
+    Lambda(Procedure, bool),
```
Comment (TyOverby) [inline src/lib.rs]: What is this bool?

**Comment by TyOverby** [comment]
Hey, looks good to me!
 
I'm still somewhat confused about how and why you changed Lambda, but feel free to merge this!

File: src/lib.rs (hunk 1 above)
Comment (bwo) [inline src/lib.rs]: boolean blindness strikes again! It's "do I get my args evaluated or not?". The design decision I'm least happy with.

**Comment by bwo** [comment]
I actually realized that macroexpansion needs to be sensitive to `quote`, because something like `'(begin ...)` shouldn't be expanded inside the `quote`. So still needs some work.

**Pull request merged**